### ISSUE TITLE
[rush] Regression after rush update: "'remote-https' is not a git command" when running "install-run-rush.js"

### ISSUE THREAD

**Issue by crhaglun** [microsoft/rushstack#2097]
**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**

After updating from Rush 5.23.2 to Rush 5.30.2, our "publish" build pipeline breaks on git errors:

````
* EXECUTING: git push origin HEAD:version/bump-1597197563662 --follow-tags --verbose --no-verify 
git: 'remote-https' is not a git command. See 'git --help'.

ERROR: The command failed with exit code 128

##[error]Cmd.exe exited with code '1'.
````

When using 5.23.2, that step is successful:

````
* EXECUTING: git push origin HEAD:version/bump-1597030761017 --follow-tags --verbose --no-verify 
POST git-receive-pack (141520 bytes)
Pushing to https://[redacted]
To https://[redacted]
 * [new branch]          HEAD -> version/bump-1597030761017
updating local tracking ref 'refs/remotes/origin/version/bump-1597030761017'
````

When attempting to reproduce locally, it indeed looks like 5.30.2 is doing something strange when executing git.
On my dev machine, running "rush version --bump" shows that git fails to execute "sh" and unexpectedly asks for a password:

````
> node common/scripts/install-run-rush.js version --bump -b user/crhaglun/version-bump-test
...
* EXECUTING: git push origin HEAD:version/bump-1597253280631 --follow-tags --verbose --no-verify
Pushing to https://msams@dev.azure.com/msams/Stream/_git/Stream.Web
error: cannot spawn sh: No such file or directory
Password for 'https://msams@dev.azure.com':^C
````
Running the git command from the same shell works as expected:
````
> git push origin HEAD:version/bump-1597253280631 --follow-tags --verbose --no-verify
Pushing to https://[redacted]
Enumerating objects: 338, done.
Counting objects: 100% (338/338), done.
Delta compression using up to 8 threads
Compressing objects: 100% (207/207), done.
Writing objects: 100% (258/258), 304.42 KiB | 2.72 MiB/s, done.
Total 258 (delta 230), reused 57 (delta 50)
POST git-receive-pack (311887 bytes)
remote: Analyzing objects... (258/258) (99 ms)
remote: Checking for credentials and other secrets... (141/141) done (1519 ms)
remote: Storing packfile... done (106 ms)
remote: Storing index... done (50 ms)
To https://[redacted]
 * [new branch]          HEAD -> version/bump-1597253280631
updating local tracking ref 'refs/remotes/origin/version/bump-1597253280631'
````

After running a bisect (update rush.json, "rush update" then "rush version --bump") it looks like 
- the last version where the git commands run as expected was 5.24.3
- the first version where the git commands fail is 5.24.4

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

This is a Microsoft-internal repo, let me know if it's meaningful to share full links to the repo and build pipelines.

**What is the expected behavior?**

We can bump versions even after updating to a newer version of rush :-)

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** Rush
* **Tool Version:** Regression from 5.24.3 to 5.24.4
* **Node Version:** 10.15.3
  * **Is this a LTS version?** Yes
  * **Have you tested on a LTS version?** Yes
* **OS:** Windows 10 (local dev machine) / Windows Server 2019 (Azure Pipelines build agent)

**Comment by acoates-ms** [comment]
I'm hitting this too.

**Comment by crhaglun** [comment]
After a round bisecting, it looks like whatever regressed happened between versions **5.24.3** and **5.24.4**

Updating description to match

**Comment by crhaglun** [comment]
Another update:

This appears to be an issue with "install-run-rush.js", i.e.

````
node common/scripts/install-run-rush.js version --bump -b master
````

hits the problem, but

````
rush version --bump -b master
````

works as expected.

**Comment by crhaglun** [comment]
Oh right, I did some further investigations here.

Adding a "SET" command to dump all environment variables in "rush.cmd" (i.e. [repo]\common\temp\install-run\@microsoft+rush@5.24.4\node_modules\.bin\rush.cmd) revealed a peculiar thing; we get _two_ path variables!

```
PATH=C:\repo\common\temp\install-run\@microsoft+rush@5.24.4\node_modules\.bin;C:\Program Files\PowerShell\7;C:\Program Files (x86)\Microsoft SDKs\VSTS\CLI\wbin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Python38\Scripts\;C:\Python38\;C:\Program Files\Common Files\Microsoft Shared\Microsoft Online Services;C:\Program Files (x86)\Common Files\Microsoft Shared\Microsoft Online Services;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\GVFS;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files (x86)\Windows Kits\10\Microsoft Application Virtualization\Sequencer\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\ManagementStudio\;C:\WINDOWS\SysWOW64\WindowsPowerShell\v1.0\Modules\TShell\TShell\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\Git\cmd;C:\ProgramData\chocolatey\bin;C:\Program Files\nodejs\;C:\Program Files (x86)\GitExtensions\;C:\Program Files\Gumshoe\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\VSShell\Common7\IDE\;C:\Program Files (x86)\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files\PowerShell\7\;C:\Users\crhaglun\AppData\Local\Microsoft\WindowsApps;C:\Users\crhaglun\AppData\Local\Dbg\UI;C:\Users\crhaglun\AppData\Local\Programs\Fiddler;C:\Program Files (x86)\DevInsights\;C:\Users\crhaglun\AppData\Local\Microsoft\WindowsApps;C:\Users\crhaglun\AppData\Local\Programs\Microsoft VS Code Insiders\bin;C:\Users\crhaglun\AppData\Roaming\npm;C:\Users\crhaglun\.dotnet\tools;C:\Users\crhaglun\AppData\Local\Programs\Microsoft VS Code\bin
Path=C:\Program Files\PowerShell\7;C:\Program Files (x86)\Microsoft SDKs\VSTS\CLI\wbin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Python38\Scripts\;C:\Python38\;C:\Program Files\Common Files\Microsoft Shared\Microsoft Online Services;C:\Program Files (x86)\Common Files\Microsoft Shared\Microsoft Online Services;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\GVFS;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files (x86)\Windows Kits\10\Microsoft Application Virtualization\Sequencer\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\ManagementStudio\;C:\WINDOWS\SysWOW64\WindowsPowerShell\v1.0\Modules\TShell\TShell\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\Git\cmd;C:\ProgramData\chocolatey\bin;C:\Program Files\nodejs\;C:\Program Files (x86)\GitExtensions\;C:\Program Files\Gumshoe\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\VSShell\Common7\IDE\;C:\Program Files (x86)\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files\PowerShell\7\;C:\Users\crhaglun\AppData\Local\Microsoft\WindowsApps;C:\Users\crhaglun\AppData\Local\Dbg\UI;C:\Users\crhaglun\AppData\Local\Programs\Fiddler;C:\Program Files (x86)\DevInsights\;C:\Users\crhaglun\AppData\Local\Microsoft\WindowsApps;C:\Users\crhaglun\AppData\Local\Programs\Microsoft VS Code Insiders\bin;C:\Users\crhaglun\AppData\Roaming\npm;C:\Users\crhaglun\.dotnet\tools;C:\Users\crhaglun\AppData\Local\Programs\Microsoft VS Code\bin
````

This does not happen in 5.23.3. I'm not sure _why_ this is causing is issues, or even _how_ this happens as environment variables should be case insensitive in Windows. A stab at how I resolved this locally is currently on reivew.

**Comment by octogonz** [comment]
I did some more investigation. Let's continue this discussion in the PR  https://github.com/microsoft/rushstack/pull/2100 comments.

**Comment by crhaglun** [comment]
I can confirm this issue has been fixed from 5.32.3 -- closing this issue.
